examples/heartbeat_demo/alicia.py

```diff
@@ -0,0 +1,158 @@
+from nucypher.characters.lawful import Bob, Ursula
+from nucypher.config.characters import AliceConfiguration
+from nucypher.config.storages import LocalFileBasedNodeStorage
+from nucypher.crypto.powers import EncryptingPower, SigningPower
+from nucypher.network.middleware import RestMiddleware
+from nucypher.utilities.logging import simpleObserver
+
+import datetime
+import os
+import shutil
+import maya
+import json
+import sys
+from twisted.logger import globalLogPublisher
+
+
+POLICY_FILENAME = "policy-metadata.json"
+
+######################
+# Boring setup stuff #
+######################
+#
+# # Twisted Logger
+globalLogPublisher.addObserver(simpleObserver)
+#
+# # Temporary file storage
+TEMP_ALICE_DIR = "{}/alicia-files".format(os.path.dirname(os.path.abspath(__file__)))
+TEMP_URSULA_CERTIFICATE_DIR = "{}/ursula-certs".format(TEMP_ALICE_DIR)
+
+# We expect the url of the seednode as the first argument.
+SEEDNODE_URL = sys.argv[1]
+
+
+#######################################
+# Alicia, the Authority of the Policy #
+#######################################
+
+
+# We get a persistent Alice.
+passphrase = "TEST_ALICIA_INSECURE_DEVELOPMENT_PASSWORD"
+try:  # If we had an existing Alicia in disk, let's get it from there
+    alice_config_file = os.path.join(TEMP_ALICE_DIR, "config_root", "alice.config")
```

Comment:
@mswilkison Caught Exception: 

```

INFO (nucypher.network.middleware.RestMiddleware): Fetching seednode https://paris-load-balancer-14b0a87b7ff37a14.elb.eu-west-3.amazonaws.com:9151 TLS certificate
Traceback (most recent call last):
  File "alicia.py", line 47, in <module>
    save_metadata=False,
  File "/Users/maclanewilkison/Projects/nucypher/nucypher/lib/python3.7/site-packages/nucypher/config/node.py", line 203, in from_configuration_file
    payload = cls._read_configuration_file(filepath=filepath)
  File "/Users/maclanewilkison/Projects/nucypher/nucypher/lib/python3.7/site-packages/nucypher/config/node.py", line 193, in _read_configuration_file
    with open(filepath, 'r') as file:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/maclanewilkison/Projects/heartbeat/examples/heartbeat_demo/alicia-files/config_root/alice.config'

```

See more: <https://gist.github.com/mswilkison/0a966d165db53b73268415771be4ab80>

Reply:
Looks like this is actually happening inside the same call stack as reading from the config file, hence the nested exception:

```
Traceback (most recent call last):
  File "alicia.py", line 59, in <module>
    minimum_stake=0)
  File "/Users/maclanewilkison/Projects/nucypher/nucypher/lib/python3.7/site-packages/nucypher/network/nodes.py", line 837, in from_seed_and_stake_info
    certificate = network_middleware.get_certificate(host=host, port=port)
  File "/Users/maclanewilkison/Projects/nucypher/nucypher/lib/python3.7/site-packages/nucypher/network/middleware.py", line 49, in get_certificate
    seednode_certificate = ssl.get_server_certificate(addr=(host, port))
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 1297, in get_server_certificate
    with  create_connection(addr) as sock:
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socket.py", line 707, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socket.py", line 748, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
```

Reply:
Related to #579